From 3ad019171933268538e16f5ab14e0f7b5f4be5e2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 8 Jan 2020 08:33:02 +1100
Subject: [PATCH] xaudio2_7: IXACT3Engine Initialize return valid error code

FACTAudioEngine_Initialize returns a uint32_t type, so we need to
convert to a valid HRESULT value.
---
 dlls/xaudio2_7/xact_dll.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dlls/xaudio2_7/xact_dll.c b/dlls/xaudio2_7/xact_dll.c
index 91e3ef0afd..1f580519d4 100644
--- a/dlls/xaudio2_7/xact_dll.c
+++ b/dlls/xaudio2_7/xact_dll.c
@@ -934,6 +934,7 @@ static HRESULT WINAPI IXACT3EngineImpl_Initialize(IXACT3Engine *iface,
 {
     XACT3EngineImpl *This = impl_from_IXACT3Engine(iface);
     FACTRuntimeParameters params;
+    UINT ret;
 
     TRACE("(%p)->(%p)\n", This, pParams);
 
@@ -972,7 +973,11 @@ static HRESULT WINAPI IXACT3EngineImpl_Initialize(IXACT3Engine *iface,
     params.fileIOCallbacks.readFileCallback = wrap_readfile;
     params.fileIOCallbacks.getOverlappedResultCallback = wrap_getoverlappedresult;
 
-    return FACTAudioEngine_Initialize(This->fact_engine, &params);
+    ret = FACTAudioEngine_Initialize(This->fact_engine, &params);
+    if (ret != 0)
+        FIXME("FACTAudioEngine_Initialize returned %d\n", ret);
+
+    return !ret ? S_OK : E_FAIL;
 }
 
 static HRESULT WINAPI IXACT3EngineImpl_ShutDown(IXACT3Engine *iface)
-- 
2.25.1

